compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureDesugar.java

```diff
@@ -808,12 +981,90 @@ public void visit(BLangIndexBasedAccess indexAccessExpr) {
         result = desugar.rewriteExpr(indexAccessExpr);
     }
 
+    public BPackageSymbol getPackageSymbol(BSymbol symbol) {
+        if (symbol.owner.getKind() == SymbolKind.PACKAGE) {
+            return (BPackageSymbol) symbol.owner;
+        }
+        return getPackageSymbol(symbol.owner);
+    }
+
     @Override
-    public void visit(BLangInvocation iExpr) {
-        iExpr.expr = rewriteExpr(iExpr.expr);
-        iExpr.requiredArgs = rewriteExprs(iExpr.requiredArgs);
-        iExpr.restArgs = rewriteExprs(iExpr.restArgs);
-        result = iExpr;
+    public void visit(BLangInvocation invocation) {
+        rewriteInvocationExpr(invocation);
+    }
+
+    public void rewriteInvocationExpr(BLangInvocation invocation) {
+        invocation.expr = rewriteExpr(invocation.expr);
+        invocation.requiredArgs = rewriteExprs(invocation.requiredArgs);
+        BInvokableTypeSymbol invokableSymbol = (BInvokableTypeSymbol) invocation.symbol.type.tsymbol;
+        if (invokableSymbol == null) {
+            invocation.restArgs = rewriteExprs(invocation.restArgs);
```

Comment:
When can `invokableSymbol` be null? And when can `invocation.restArgs` not be null when `invokableSymbol` is null?

Reply:
Most of the usage of this constructor(https://github.com/ballerina-platform/ballerina-lang/blob/54cd902acf49507a67f473c1350a250935959d0a/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BInvokableType.java#L46) set type symbol as null.